src/rust/engine/fs/src/glob_matching.rs

```diff
@@ -313,7 +322,9 @@ trait GlobMatchingImplementation<E: Send + Sync + 'static>: VFS<E> {
           } else {
             // TODO(#5683): this doesn't have any useful context (the stack trace) without
             // being thrown -- this needs to be provided, otherwise this is unusable.
-            warn!("{}", msg);
+            // NB: warn!() is blocked when using a console task e.g. list, so we print
+            // unconditionally to stderr here.
+            eprintln!("WARN] {}", msg);
```

Comment:
This change isn't related to the rest of the PR, and should probably land separately. It significantly affects the verbosity of pantsd runs.

Reply:
0aa2db394db13553e436531c060d08ae1e9d13ac reverts this. I will make a new issue in a second to track the underlying problem.

Reply:
I also think...there may not even need to be a followup issue, as all the tests are passing without any being skipped? Will try to gain some more clarity.